src/chunkserver/network_worker_thread.cc

```diff
@@ -1083,90 +1127,99 @@ void worker_gotpacket(csserventry *eptr, uint32_t type, const uint8_t *data, uin
 	TRACETHIS();
 	uint32_t type, size;
 	const uint8_t *ptr;
-	if (eptr->state == WRITEFWD) {
-		if (eptr->mode == DATA && eptr->inputpacket.bytesleft == 0 && eptr->fwdbytesleft == 0) {
-			ptr = eptr->hdrbuff;
+	if (eptr->state == ChunkserverEntry::State::WriteForward) {
+		if (eptr->mode == ChunkserverEntry::Mode::Data &&
+		    eptr->inputPacket.bytesLeft == 0 && eptr->fwdBytesLeft == 0) {
+			ptr = eptr->headerBuffer;
 			type = get32bit(&ptr);
 			size = get32bit(&ptr);
 
-			eptr->mode = HEADER;
-			eptr->inputpacket.bytesleft = 8;
-			eptr->inputpacket.startptr = eptr->hdrbuff;
+			eptr->mode = ChunkserverEntry::Mode::Header;
+			eptr->inputPacket.bytesLeft = PacketHeader::kSize;
+			eptr->inputPacket.startPtr = eptr->headerBuffer;
 
-			worker_gotpacket(eptr, type, eptr->inputpacket.packet + 8, size);
+			worker_gotpacket(eptr, type,
+			                 eptr->inputPacket.packet + PacketHeader::kSize,
+			                 size);
 
-			if (eptr->inputpacket.packet) {
-				free(eptr->inputpacket.packet);
+			if (eptr->inputPacket.packet) {
+				free(eptr->inputPacket.packet);
 			}
-			eptr->inputpacket.packet = NULL;
+			eptr->inputPacket.packet = nullptr;
 		}
 	} else {
-		if (eptr->mode == DATA && eptr->inputpacket.bytesleft == 0) {
-			ptr = eptr->hdrbuff;
+		if (eptr->mode == ChunkserverEntry::Mode::Data &&
+		    eptr->inputPacket.bytesLeft == 0) {
+			ptr = eptr->headerBuffer;
 			type = get32bit(&ptr);
 			size = get32bit(&ptr);
 
-			eptr->mode = HEADER;
-			eptr->inputpacket.bytesleft = 8;
-			eptr->inputpacket.startptr = eptr->hdrbuff;
+			eptr->mode = ChunkserverEntry::Mode::Header;
+			eptr->inputPacket.bytesLeft = PacketHeader::kSize;
+			eptr->inputPacket.startPtr = eptr->headerBuffer;
 
-			worker_gotpacket(eptr, type, eptr->inputpacket.packet, size);
+			worker_gotpacket(eptr, type, eptr->inputPacket.packet, size);
 
-			if (eptr->inputpacket.packet) {
-				free(eptr->inputpacket.packet);
+			if (eptr->inputPacket.packet) {
+				free(eptr->inputPacket.packet);
 			}
-			eptr->inputpacket.packet = NULL;
+			eptr->inputPacket.packet = nullptr;
 		}
 	}
 }
 
-void worker_fwdconnected(csserventry *eptr) {
+void worker_fwdconnected(ChunkserverEntry *eptr) {
 	TRACETHIS();
 	int status;
-	status = tcpgetstatus(eptr->fwdsock);
+	status = tcpgetstatus(eptr->fwdSocket);
 	if (status) {
 		safs_silent_errlog(LOG_WARNING, "connection failed, error");
 		worker_fwderror(eptr);
 		return;
 	}
-	tcpnodelay(eptr->fwdsock);
-	eptr->state = WRITEINIT;
+	tcpnodelay(eptr->fwdSocket);
+	eptr->state = ChunkserverEntry::State::WriteInit;
 }
 
-void worker_fwdread(csserventry *eptr) {
+void worker_fwdread(ChunkserverEntry *eptr) {
 	TRACETHIS();
 	int32_t i;
 	uint32_t type, size;
 	const uint8_t *ptr;
-	if (eptr->fwdmode == HEADER) {
-		i = read(eptr->fwdsock, eptr->fwdinputpacket.startptr,
-				eptr->fwdinputpacket.bytesleft);
+	if (eptr->fwdMode == ChunkserverEntry::Mode::Header) {
+		i = read(eptr->fwdSocket, eptr->fwdInputPacket.startPtr,
```

Comment:
Maybe it's worth to rename `i` for a more meaningful name like `readBytes` and also change it's type to `ssize_t` to avoid warning about `Narrowing conversion from ssize_t to int32_t`.

Reply:
Applied here: https://github.com/leil-io/saunafs/commit/8f8dcf733ccd489fa6b9b4426b1ec5629500c839
Please notice this is out of the scope of this change.